src/sagemaker/serve/validations/optimization.py

```diff
@@ -0,0 +1,229 @@
+        """Validator for optimization containers"""
+
+        # check the validity of each individual field
+        if not optimization_combination.compilation.issubset(self.compilation):
+            raise ValueError("Compilation")
+        if not optimization_combination.quantization_technique.issubset(
+            self.quantization_technique
+        ):
+            copy_quantization_technique = optimization_combination.quantization_technique.copy()
+            raise ValueError(f"Quantization:{copy_quantization_technique.pop()}")
+        if not optimization_combination.speculative_decoding.issubset(self.speculative_decoding):
+            raise ValueError("Speculative Decoding")
+        if not optimization_combination.sharding.issubset(self.sharding):
+            raise ValueError("Sharding")
+
+        # optimization technique combinations that need to be validated
+        if optimization_combination.compilation and optimization_combination.speculative_decoding:
+            is_compiled = optimization_combination.compilation.copy().pop()
+            is_speculative_decoding = optimization_combination.speculative_decoding.copy().pop()
+            if is_compiled and is_speculative_decoding:
+                raise ValueError("Compilation and Speculative Decoding together")
+
+        if rule_set == _OptimizationContainer.TRT:
+            is_compiled = optimization_combination.compilation.copy().pop()
+            is_quantized = optimization_combination.quantization_technique.copy().pop()
+            if is_quantized and not is_compiled:
+                raise ValueError(f"Quantization:{is_quantized} must be provided with Compilation")
+
+
+TRUTHY_SET = {None, True}
+FALSY_SET = {None, False}
+TRT_CONFIGURATION = {
+    "supported_instance_families": {"p4d", "p4de", "p5", "g5", "g6"},
+    "optimization_combination": _OptimizationCombination(
+        optimization_container=_OptimizationContainer.TRT,
+        compilation=TRUTHY_SET,
+        quantization_technique={None, "awq", "fp8", "smoothquant"},
+        speculative_decoding=FALSY_SET,
+        sharding=FALSY_SET,
+    ),
+}
+VLLM_CONFIGURATION = {
+    "supported_instance_families": {"p4d", "p4de", "p5", "g5", "g6"},
+    "optimization_combination": _OptimizationCombination(
+        optimization_container=_OptimizationContainer.VLLM,
+        compilation=FALSY_SET,
+        quantization_technique={None, "awq", "fp8"},
+        speculative_decoding=TRUTHY_SET,
+        sharding=TRUTHY_SET,
+    ),
+}
+NEURON_CONFIGURATION = {
+    "supported_instance_families": {"inf2", "trn1", "trn1n"},
+    "optimization_combination": _OptimizationCombination(
+        optimization_container=_OptimizationContainer.NEURON,
+        compilation=TRUTHY_SET,
+        quantization_technique={None},
+        speculative_decoding=FALSY_SET,
+        sharding=FALSY_SET,
+    ),
+}
+
+
+def _validate_optimization_configuration(
```

Comment:
this is such a big method, we should look to optimise this?

Reply:
also I don't see any unit tests coverage for functionality in this module?

Reply:
Should be able to view the UTs here: https://github.com/aws/sagemaker-python-sdk/pull/4927/files#diff-5f58a16d071696e26bf4081f3f01294336f6c669407a9474ed7c0747af0e629fR3479. and Ack, we will look to optimize that